clusterloader2/pkg/measurement/common/network/network_performance_measurement.go

```diff
@@ -0,0 +1,449 @@
+// scale (N:M) pod ratios.Client and server pods located on different worker
+// nodes exchange traffic for specified time to measure the performance metrics.
+package network
+
+import (
+	"container/heap"
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog"
+	"k8s.io/perf-tests/clusterloader2/pkg/framework"
+	"k8s.io/perf-tests/clusterloader2/pkg/framework/client"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement"
+	measurementutil "k8s.io/perf-tests/clusterloader2/pkg/measurement/util"
+	"k8s.io/perf-tests/clusterloader2/pkg/util"
+)
+
+const (
+	podReadyTimeOut = 5 * time.Minute
+	// initialDelay in seconds after which pod starts sending traffic.
+	// Delay is used to synchronize all client pods to send traffic at same time.
+	initialDelayForTestExecution = 5 * time.Second
+	// networkPerformanceMetricsName indicates the measurement name
+	networkPerformanceMetricsName = "NetworkPerformanceMetrics"
+	netperfNamespace              = "netperf"
+)
+
+const (
+	manifestPathPrefix                  = "$GOPATH/src/k8s.io/perf-tests/clusterloader2/pkg/measurement/common/network/manifests"
+	workerPodDeploymentManifestFilePath = manifestPathPrefix + "/" + "*deployment.yaml"
+	networkTestRequestFilePath          = manifestPathPrefix + "/" + "networktestrequests.yaml"
+	crdManifestFilePath                 = manifestPathPrefix + "/" + "*CustomResourceDefinition.yaml"
+	clusterRoleBindingFilePath          = manifestPathPrefix + "/" + "roleBinding.yaml"
+	customResourceDefinitionName        = "networktestrequests.clusterloader.io"
+	rbacName                            = "networktestrequests-rbac"
+)
+
+var (
+	crdGvk = schema.GroupVersionKind{
+		Group:   "apiextensions.k8s.io",
+		Kind:    "CustomResourceDefinition",
+		Version: "v1",
+	}
+
+	rbacGvk = schema.GroupVersionKind{
+		Group:   "rbac.authorization.k8s.io",
+		Kind:    "ClusterRoleBinding",
+		Version: "v1",
+	}
+
+	crGvk = schema.GroupVersionKind{
+		Group:   "clusterloader.io",
+		Kind:    "NetworkTestRequest",
+		Version: "v1alpha1",
+	}
+)
+
+func init() {
+	klog.Info("Registering Network Performance Measurement")
+	if err := measurement.Register(networkPerformanceMetricsName, createNetworkPerformanceMeasurement); err != nil {
+		klog.Fatalf("Cannot register %s: %v", networkPerformanceMetricsName, err)
+	}
+}
+
+func createNetworkPerformanceMeasurement() measurement.Measurement {
+	return &networkPerformanceMeasurement{}
+}
+
+type networkPerformanceMeasurement struct {
+	k8sClient         kubernetes.Interface
+	dynamicClient     dynamic.Interface
+	resourceInterface dynamic.ResourceInterface
+	framework         *framework.Framework
+	numberOfServers   int
```

Comment:
nit: add empty line before "numberOfServer"

[this splits a different groups of parameters]

Reply:
modified